starter-code/src/functions-and-arrays.js

```diff
@@ -98,3 +164,37 @@ var matrix = [
   [20, 73, 35, 29, 78, 31, 90, 1, 74, 31, 49, 71, 48, 86, 81, 16, 23, 57, 5, 54],
   [1, 70, 54, 71, 83, 51, 54, 69, 16, 92, 33, 48, 61, 43, 52, 1, 89, 19, 67, 48]
 ];
+
+function greatestProduct(numMatrix) {  
+  var greatestProductArray = [];  
+  var greatestProductCandidate = 0;
+  
+  // Horizontal greatest products in each row
+  for (var i = 0; i < numMatrix.length; i++) {    
+    for (var j = 0; j < numMatrix[i].length - 4; j++) {
+      var currentProduct = numMatrix[i][j] * numMatrix[i][j+1] * numMatrix[i][j+2] * numMatrix[i][j+3];
+      if (currentProduct > greatestProductCandidate)
+        greatestProductCandidate = currentProduct;
+    }
+    greatestProductArray.push(greatestProductCandidate);
+  }
+  
+  // Vertical greatest products in each column
+  greatestProductCandidate = 0;  
+  for (var i = 0; i < numMatrix[0].length; i++) {        
+    for (var j = 0; j < numMatrix.length - 4; j++) {
+      var currentProduct = numMatrix[j][i] * numMatrix[j+1][i] * numMatrix[j+2][i] * numMatrix[j+3][i];
+      if (currentProduct > greatestProductCandidate)
+        greatestProductCandidate = currentProduct;
+    }
+    greatestProductArray.push(greatestProductCandidate);
+  }  
+
+  // I get the greatest product of the array
+  greatestProductCandidate = 0;
+  for (var i = 0; i < greatestProductArray.length; i++) {
+    greatestProductCandidate = maxOfTwoNumbers(greatestProductCandidate, greatestProductArray[i]);
+  }
+
```

Comment:
Está bien pensado, pero te complicas bastante, cómo lo harías más sencillo? Ten en cuenta que en el ejemplo solo os dan un array de dos dimensiones de 20 x 20. Y luego, lo habéis probado con un array pequeño de 4 x 4? Es la manera más sencilla porque podéis calcularlo a mano y cercioraros de que está bien.